01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+
+```{r}
+#| label: load-acs
+#| cache: true
+#| dependson: "import-acs-one-year"
+#| include: false
+
+# Set microdata to whatever the file you downloaded in the previous chunk is called
+microdata <- "usa_00010.xml"
+
+acs1yr_raw <-
+  read_ipums_micro(here(glue("01_financial-well-being/data/{microdata}"))) |>
+  janitor::clean_names() |>
+  rename(
+    ownership = ownershp,
+    state = statefip,
+    county = countyfip,
+  ) |>
+  filter(
+    year != 2020,
+    pernum == 1
+  )
+```
+
+The [ipumsr](https://tech.popdata.org/ipumsr/articles/value-labels.html#remove-unused-value-labels) documentation says that:
+
+> "labelled vectors are not intended for use throughout the entire analysis process. Instead, they should be used during the initial data preparation process to convert raw data into values that are more meaningful. These can then be converted to other variable types (often factors) for analysis. Unfortunately, this isn’t a process that can typically be automated, as it depends primarily on the research questions the data will be used to address. However, ipumsr provides several functions to manipulate value labels to make this process easier.
+
+A useful way of finding missing value labels is to call `ipums_var_info(acs1yr)` and inspect the `val_labels`.
+
+Notice that missing values for "value of housing units" (`valueh`) are labelled as 9999998, and 9999999. We fix this before doing any computations.
+
+```{r}
+#| label: handle-missingness
+
+
+acs1yr <-
+  acs1yr_raw |>
+  mutate(
+    valueh = lbl_na_if(valueh, ~ .lbl %in% c("Missing", "N/A")),
+  )
+
+acs1yr |>
+  assertr::verify(valueh < 9999998 | is.na(valueh))
+```
+
+Next, we exclude individuals living in group quarters because we only want households for this metric. In particular, we exclude people living in places other than housing units (e.g., correctional facilities, nursing homes, mental hospitals, dorms, barracks, group homes, missions). For more information, see the [variable description](https://usa.ipums.org/usa-action/variables/GQ#codes_section) on IPUMS USA.
+
+```{r}
+#| label: keep-households-only
+acs1yr <-
+  acs1yr |>
+  filter(!(gq %in% c("Group quarters--Institutions", "Other group quarters")))
+```
+
+FIPS codes should always contain leading zeros so that state codes are two digits, county codes are three digits, and place codes are five digits.
+
+```{r}
+#| label: clean-fips-codes
+
+acs1yr <-
+  acs1yr |>
+  mutate(
+    across(c(state, county, city, puma), as.character),
+    state = str_pad(state, width = 2, pad = "0", side = "left"),
+    county = str_pad(county, width = 3, pad = "0", side = "left"),
+    city = str_pad(city, width = 5, pad = "0", side = "left"),
+    puma = str_pad(puma, width = 5, pad = "0", side = "left")
+  )
+```
```

Comment:
After running this code chunk, I look at acs1yr and these variables look numeric, not string and do not have leading zeros.